discojs/src/training/trainer.ts

```diff
@@ -38,6 +52,9 @@ export class Trainer<D extends DataType, N extends Network> {
     AsyncGenerator<AsyncGenerator<BatchLogs, EpochLogs>, RoundLogs>,
     void
   >;
+  // Map of weight Index and weight update
+  weightHistory : WeightHistory = new Map<number, number[]>();
+  previousEpochWeights: WeightsContainer | undefined = undefined;
```

Comment:
class variables defined this way make theses public, better to use the [somewhat-new-but-very-useful private elements](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_elements) to keep theses private to the class.

also, you don't need to set `undefined`, it has this value by default

```suggestion
  #weightHistory : WeightHistory = new Map<number, number[]>();
  #previousEpochWeights: WeightsContainer | undefined;
```